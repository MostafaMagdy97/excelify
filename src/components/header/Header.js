import React from 'react';
import { Flex } from 'grid-styled';
import styled from 'styled-components';

import Text from '../text/Text';
import Button from '../buttons/Button';
import { COLORS_VALUES, COLORS } from '../../base/Colors';
import Space from '../../base/Space';
import { FONT_TYPES } from '../../base/Typography';

const HeaderContainer = Flex.extend`
  border-bottom: 1px solid ${COLORS_VALUES[COLORS.BORDER]};
  padding: ${Space[2]}px ${Space[4]}px;
`;

const Logo = styled(Text)`
  font-family: 'Roboto Slab';
  font-weight: 600;
  color: ${COLORS_VALUES[COLORS.BLUE]};
`;

const Header = props => (
  <HeaderContainer alignItems="center" justifyContent="space-between" {...props}>
    <Logo tag="h1" key="logo" type={FONT_TYPES.TITLE} m="0">
      Cerberus
    </Logo>
    <Button>Convert Image</Button>
  </HeaderContainer>
);

export default Header;
