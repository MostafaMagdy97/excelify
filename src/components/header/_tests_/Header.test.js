import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import 'jest-styled-components';

import Header from '../Header';

configure({ adapter: new Adapter() });

describe('Header', () => {
  it('should render', () => {
    const tree = render(<Header />);
    expect(tree).toMatchSnapshot();
  });
});
