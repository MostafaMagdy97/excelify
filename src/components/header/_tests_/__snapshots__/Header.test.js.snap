// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
.c2 {
  margin: 0;
  color: text;
  font-weight: 500;
  font-size: 21px;
  line-height: 24px;
  font-display: fallback;
}

.c4 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: blue;
  font-weight: 500;
  font-size: 15px;
  line-height: 24px;
  z-index: 1;
  font-display: fallback;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 32px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
  background-color: transparent;
}

.c3:not(:disabled) {
  cursor: pointer;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid #f5f5f5;
  padding: 8px 32px;
}

.c1 {
  font-family: 'Roboto Slab';
  font-weight: 600;
}

@media screen and (min-width:40em) {
  .c2 {
    font-size: 21px;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    font-size: 20px;
  }
}

@media screen and (min-width:64em) {
  .c2 {
    font-size: 20px;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    font-size: 15px;
  }
}

@media screen and (min-width:52em) {
  .c4 {
    font-size: 14px;
  }
}

@media screen and (min-width:64em) {
  .c4 {
    font-size: 14px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1 c2"
    color="text"
    font-size="21px,21px,20px,20px"
    font-weight="500"
    lineheight="24px"
    m="0"
    tag="h1"
    type="title"
  >
    Cerberus
  </h1>
  <button
    class="c3"
    color="blue"
  >
    <span
      class="c4"
      color="blue"
      font-size="15px,15px,14px,14px"
      font-weight="500"
      lineheight="24px"
      tag="span"
      type="body"
      width="max-content"
      zindex="1"
    >
      Upload a new image
    </span>
  </button>
</div>
`;
